### PR TITLE
Simplify C# print methods

### PR DESCRIPTION
**Pull request by raulsntos**
- Extracts the parameters logic to a single method
- Simplify the handling of null parameters

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 using System;
 using System.Collections.Generic;
 using System.Runtime.CompilerServices;
+using System.Diagnostics.CodeAnalysis;
```
Comment (aaronfranke) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs:9]:
```suggestion
using System.Diagnostics.CodeAnalysis;
using System.Runtime.CompilerServices;
```

Let's keep these in alphabetical order at least.

**Review by aaronfranke — CHANGES_REQUESTED** [review]
Overall this is nice. Having this logic be in a separate method is better than the repeated one-liners.

I tested this cherry-picked on 3.x and it gives me this error:

```
Core/GD.cs(85,49): error CS0246: The type or namespace name 'AllowNullAttribute' could not be found (are you missing a using directive or an assembly reference?) [/home/aaronfranke/workspace/godot-3/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]                                                                               
Core/GD.cs(85,49): error CS0246: The type or namespace name 'AllowNull' could not be found (are you missing a using directive or an assembly reference?) [/home/aaronfranke/workspace/godot-3/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
```

Removing `[AllowNull]` made this go away. This is likely just something to keep in mind when cherry-picking (and then the `using` isn't necessary).

Generally I'd like to keep C# methods in alphabetical order unless there is a significant reason to not do so. In this case this method is very related to the print methods so that may be enough of a reason. We can either move `GetPrintParams` above `Hash` or just leave it where it is, I don't have super strong feelings either way but I'm leaning towards moving it.

**Comment by raulsntos** [comment]
That's because in the `3.x` branch the `TargetFramework` is `netstandard2.0` while in `master` it's `netstandard2.1`.

**3.x**
https://github.com/godotengine/godot/blob/d9a07d1528a438ddf97b1a9cd37a96cbfa3a92ce/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj#L7

**master**
https://github.com/godotengine/godot/blob/769de0c35f10780edc1a670b09f63fc730fefc39/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj#L7

The `AllowNullAttribute` is only available from `netstandard2.1`:
https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.allownullattribute?view=net-5.0

I guess I could remove it to make it easier to cherry-pick but at one point I'd like to annotate the  API with [nullable](https://devblogs.microsoft.com/dotnet/embracing-nullable-reference-types/) attributes to allow users who enable it in their projects get warnings and such.

**Comment by aaronfranke** [comment]
IIRC @neikeq said that he doesn't want the 3.x and master codebases to diverge too much for the time being.

**Review by aaronfranke — APPROVED** [review]

**Comment by raulsntos** [comment]
I ended up removing it, we can annotate the API later.

**Comment by neikeq** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Pull request merged**